application/comit_node/src/swap_protocols/rfc003/secret.rs

```diff
@@ -6,8 +6,35 @@ use std::{
     str::FromStr,
 };
 
+#[derive(PartialEq, Debug)]
+pub enum FromErr {
+    InvalidLength { expected: usize, got: usize },
+    FromHex(hex::FromHexError),
+}
+
 #[derive(Clone, PartialOrd, Ord, PartialEq, Eq, Hash)]
-pub struct SecretHash(pub Vec<u8>);
+pub struct SecretHash([u8; Self::LENGTH]);
+
+impl SecretHash {
+    pub const LENGTH: usize = 32;
+
+    pub fn from_vec(vec: &[u8]) -> Result<SecretHash, FromErr> {
```

Comment:
❓Can't you make `vec: &[u8]` already a fixe length array and get rid of the check below?